### PR TITLE
Fixed an issue where updatejs was not updating due to redirect 302

### LINKED ISSUE #262: updatedb.js: ERROR: HTTP Request Failed [302 Found]
Hello, 
I have an 302 error when updating the database using `node ./node_modules/geoip-lite/scripts/updatedb.js license_key=MY_LICENSE_KEY`

A check on the browser's network showed that there was a redirection from `https://download.maxmind.com/app/geoip_download...` to Maxmind R2 presigned URLs `mm-prod-geoip-databases.a2649acb697e2c09b632799562c076f2.r2.cloudflarestorage.com`

I think there is a need to add a redirection code for the HTTP request
 `if ([301, 302, 303, 307, 308].includes(status)) {
  var newLocation = response.headers['location'];
  console.log('Redirecting to', newLocation);
  client = https.get(newLocation, onResponse); 
  return;}`

### PR DESCRIPTION
**Pull request by JamieStivala**
Follows the redirect on MaxMind Servers to prevent issue "Error 302"

Fixes #262

### CODE REVIEW

File: scripts/updatedb.js [hunk 1]
```diff
@@ -181,8 +181,10 @@ function check(database, cb) {
         
 		function onResponse(response) {
 			var status = response.statusCode;
-    
-			if (status !== 200) {
+
+			if(status === 302 || status === 301){
```
Comment (bluesmoon) [inline scripts/updatedb.js:185]: can you add 303, 307 and 308 as well?

Comment (bluesmoon) [reply scripts/updatedb.js:185]: also, please follow the coding style - space before `(` and after `)`

File: scripts/updatedb.js [hunk 2]
```diff
@@ -181,8 +181,10 @@ function check(database, cb) {
...
+				return https.get(getHTTPOptions(response.headers.location), onResponse);
+			}else if (status !== 200) {
```
Comment (bluesmoon) [inline scripts/updatedb.js:187]: same with the coding style here, put a space after the `}`

**Comment by bluesmoon** [comment]
Thanks for the PR. Left a few comments.

**Comment by JamieStivala** [comment]
Thanks for quickly reviewing. Updated the PR to match the code style and added redirect 303, 307 and 308.

**Comment by justColbs** [comment]
I see this was merged into main, but it's not coming down with npm install geoip-lite. I can download from the branch or the commit, but it takes sooooo long for some reason. Any idea when this will be available via npm? Thanks!

**Comment by JamieStivala** [comment]
The change has not been pushed to the NPM Yet. What you can do until @bluesmoon pushed it to NPM is install the package from GitHub by doing `npm install git+https://github.com/geoip-lite/node-geoip.git`

**Comment by bluesmoon** [comment]
Hi all, I've pushed the new version to npm.

**Comment by justColbs** [comment]
@bluesmoon Thank you so much! Works great now. I wanted to mention that Github and NPM are not aligned yet it appears. unless I declare @1.4.10 at install this update doesn't install this redirect error fix. Just thought I'd mention because I'm sure you plan to update and didn't want anyone else to get confused.

**Comment by bluesmoon** [comment]
@justColbs not sure what that means. npm shouldn't be pulling from github. When I look at https://www.npmjs.com/package/geoip-lite it shows that 1.4.10 is the latest version.

The git code has already moved past 1.4.10 since I'm making changes to the CI workflow.

**Comment by bluesmoon** [comment]
I checked the code that's cached in npm, and it shows the redirect handling code in there.

**Pull request merged**